packages/js/src/ui/components/PopoverContent.tsx

```diff
@@ -0,0 +1,55 @@
+import { createMemo, JSX, onCleanup, onMount, ParentComponent, Show } from 'solid-js';
+import { Portal } from 'solid-js/web';
+import { getOffsetTranslate, getPositionClasses, useStyle } from '../helpers';
+import { usePopover } from './Popover';
+
+export const PopoverContent: ParentComponent = (props) => {
+  const style = useStyle();
+  const { setContentRef, opened, targetRef, position, offset, onClose, contentRef } = usePopover();
+
+  const handleClickOutside = (e: any) => {
+    if (contentRef()?.contains(e.target)) return;
+    onClose();
+  };
+
+  const handleEscapeKey = (e: any) => {
```

Comment:
❓ Why is this and outside click handling needed? https://developer.mozilla.org/en-US/docs/Web/API/Popover_API/Using#auto_state_and_light_dismiss

Reply:
Good call, let's remove if not necessary. Does it maybe have to do with Solid, knowing the state of the popover div?